pincer/objects/app/interactions.py

```diff
@@ -142,61 +143,65 @@ class Interaction(APIObject):
     def __post_init__(self):
         super().__post_init__()
 
-        self._convert_functions = {
-            AppCommandOptionType.SUB_COMMAND: None,
-            AppCommandOptionType.SUB_COMMAND_GROUP: None,
-
-            AppCommandOptionType.STRING: str,
-            AppCommandOptionType.INTEGER: int,
-            AppCommandOptionType.BOOLEAN: bool,
-            AppCommandOptionType.NUMBER: float,
-
-            AppCommandOptionType.USER: lambda value:
-            self._client.get_user(
-                convert(value, Snowflake.from_string)
-            ),
-            AppCommandOptionType.CHANNEL: lambda value:
-            self._client.get_channel(
-                convert(value, Snowflake.from_string)
-            ),
-            AppCommandOptionType.ROLE: lambda value:
-            self._client.get_role(
-                convert(self.guild_id, Snowflake.from_string),
-                convert(value, Snowflake.from_string)
-            ),
-            AppCommandOptionType.MENTIONABLE: None
-        }
-
-    async def build(self):
-        """|coro|
-
-        Sets the parameters in the interaction that need information
-        from the discord API.
+        for option in self.data.options:
+
+            if option.type is AppCommandOptionType.STRING:
```

Comment:
does this big `elif` chain is actually any faster?

Reply:
No idea. It's there now since Mentionable runs two different functions so a dictionary doesn't work.
The optimazatiom comes from me removing unnecessary request to the discord API.